compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1629,6 +1627,21 @@ macro_rules! nop_lift {
     };
 }
 
+// Can't use the macros as we have reuse the `substs` here.
+impl<'a, 'tcx> Lift<'tcx> for &'a List<Ty<'a>> {
+    type Lifted = &'tcx List<Ty<'tcx>>;
+    fn lift_to_tcx(self, tcx: TyCtxt<'tcx>) -> Option<Self::Lifted> {
```

Comment:
Given the transmute inside it, in instances such as this one, I personally like to spell out the aliases explicitly:

```suggestion
    fn lift_to_tcx(self: &'a List<Ty<'a>>, tcx: TyCtxt<'tcx>) -> Option<&'tcx List<Ty<'tcx>>> {
```